### PR TITLE
UI: connect to wifi prompt

### PR DESCRIPTION
**Pull request by incognitojam**
wifi widget

- move PrimeAd to left column, replacing DriveStats when user is not subscribed to prime (expected to be replaced by the MapPanel in https://github.com/commaai/openpilot/pull/27629)
- shrink comma prime widget
- add "Setup Wi-Fi" widget
- switch between setup/uploading using network state

||New|Old|
|-|-|-|
|Unpaired|||
|No prime (Wi-Fi)|||
|No prime (No Wi-Fi)||
|Prime|||
|Prime (No Wi-Fi)||

### CODE REVIEW

**Comment by sshane** [comment]
Did you try making the button expand to fill the width like the pair button?

**Comment by incognitojam** [comment]
Style=Setup Wi-Fi
Like this?

File: selfdrive/ui/qt/widgets/wifi.cc [hunk 1]
```diff
@@ -0,0 +1,103 @@
...
+    icon->setPixmap(pixmap->scaledToWidth(100, Qt::SmoothTransformation));
+    uploading_layout->addWidget(icon, 0, Qt::AlignHCenter);
+
+    QLabel *title = new QLabel(tr("Uploading your training data"));
+    title->setAlignment(Qt::AlignHCenter);
+    title->setStyleSheet("font-size: 64px; font-weight: 600;");
+    title->setWordWrap(true);
+    uploading_layout->addWidget(title, 0, Qt::AlignHCenter);
+
+    QLabel *desc = new QLabel(tr("Your driving data helps improve openpilot"));
```
Comment (incognitojam) [inline selfdrive/ui/qt/widgets/wifi.cc:77]: Text looks a bit small on device

File: selfdrive/ui/qt/home.cc [hunk 2]
```diff
@@ -136,21 +137,34 @@ OffroadHome::OffroadHome(QWidget* parent) : QFrame(parent) {
     home_layout->setContentsMargins(0, 0, 0, 0);
     home_layout->setSpacing(30);
 
-    // left: ExperimentalModeButton, DriveStats
-    QWidget* left_widget = new QWidget(this);
-    QVBoxLayout* left_column = new QVBoxLayout(left_widget);
-    left_column->setContentsMargins(0, 0, 0, 0);
-    left_column->setSpacing(30);
+    // left: DriveStats, PrimeAdWidget
```
Comment (sshane) [inline selfdrive/ui/qt/home.cc:140]:
```suggestion
    // left: DriveStats/PrimeAdWidget
```

File: selfdrive/ui/qt/widgets/wifi.cc [hunk 3]
```diff
@@ -0,0 +1,103 @@
...
+    {
+      QLabel *title = new QLabel(tr("Uploading training data"));
+      title->setStyleSheet("font-size: 64px; font-weight: 600;");
+      title->setWordWrap(true);
+      title->setSizePolicy(QSizePolicy::Expanding, QSizePolicy::Minimum);
+      title_layout->addWidget(title);
+      title_layout->addStretch();
+
+      QLabel *icon = new QLabel;
+      QPixmap *pixmap = new QPixmap("../assets/offroad/icon_wifi_uploading.svg");
```
Comment (sshane) [inline selfdrive/ui/qt/widgets/wifi.cc:71]: should add to assets.qrc and use ASSETS_PATH. this doesn't work if you're not in `selfdrive/ui`, right?

File: selfdrive/ui/qt/widgets/wifi.cc [hunk 4]
```diff
@@ -0,0 +1,103 @@
...
+      icon->setPixmap(pixmap->scaledToWidth(120, Qt::SmoothTransformation));
+      title_layout->addWidget(icon);
+    }
+    uploading_layout->addLayout(title_layout);
+
+    QLabel *desc = new QLabel(tr("Your data is used to train driving models and helps improve openpilot"));
```
Comment (sshane) [inline selfdrive/ui/qt/widgets/wifi.cc:77]:
```suggestion
    QLabel *desc = new QLabel(tr("Your data is used to train driving models and help improve openpilot"));
```

"your data...helps improve openpilot"
or
"your data is used to...help improve openpilot"

I read the second first.

image

But now I'm second guessing myself

**Pull request merged**